Build.helpers.psm1

```diff
@@ -0,0 +1,814 @@
+        objects for common APPX manifest elements.
+
+    .PARAMETER Path
+        The full path to the APPX manifest file to load.
+
+    .EXAMPLE
+        $manifest = Get-XmlElement -Path "C:\src\Package.appxmanifest"
+        $identity = $manifest.Document.Root.Element($manifest.Identity)
+
+    .OUTPUTS
+        PSCustomObject
+        Returns an object with the following properties:
+        - Document: The loaded XDocument
+        - Identity: XName for Identity element
+        - Properties: XName for Properties element
+        - DisplayName: XName for DisplayName element
+        - Applications: XName for Applications element
+        - Application: XName for Application element
+        - VisualElements: XName for VisualElements element
+    #>
+    [CmdletBinding()]
+    [OutputType([PSCustomObject])]
+    param (
+        [Parameter(
+            Mandatory = $true,
+            Position = 0,
+            ValueFromPipeline = $true,
+            ValueFromPipelineByPropertyName = $true,
+            HelpMessage = "Path to the APPX manifest file"
+        )]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({
+                if (Test-Path $_ -PathType Leaf)
+                {
+                    $true
+                }
+                else
+                {
+                    throw "The specified path '$_' does not exist or is not a file."
+                }
+            })]
+        [string]$Path
+    )
+
+    process
+    {
+        Write-Verbose "Loading XML from path: $Path"
+
+        [void][Reflection.Assembly]::LoadWithPartialName("System.Xml.Linq")
+
+        try
+        {
+            $appxmanifest = [System.Xml.Linq.XDocument]::Load($Path)
+            Write-Verbose "Successfully loaded XML document"
+        }
+        catch
+        {
+            throw "Failed to load XML document from '$Path': $_"
+        }
+
+        $namespace = "http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+        $uapNamespace = "http://schemas.microsoft.com/appx/manifest/uap/windows10"
+        $result = [PSCustomObject]@{
+            Document       = $appxmanifest
+            Identity       = [System.Xml.Linq.XName]::Get("{$namespace}Identity")
+            Properties     = [System.Xml.Linq.XName]::Get("{$namespace}Properties")
+            DisplayName    = [System.Xml.Linq.XName]::Get("{$namespace}DisplayName")
+            Applications   = [System.Xml.Linq.XName]::Get("{$namespace}Applications")
+            Application    = [System.Xml.Linq.XName]::Get("{$namespace}Application")
+            VisualElements = [System.Xml.Linq.XName]::Get("{$uapNamespace}VisualElements")
+        }
+
+        return $result
+    }
+}
+
+function Restore-NuGet
+{
+    [CmdletBinding()]
+    param (
+        [string]$SolutionPath,
+        [switch]$UseInternal
+    )
+
+    $nugetConfig = Join-Path (Split-Path $SolutionPath -Parent) "nuget.config"
```

Comment:
The pipeline is complaining about the nuget.config containing more than one source. I believe we need to apply this step at the very beginning of the pipeline